lib/puma/server.rb

```diff
@@ -360,7 +362,10 @@ def handle_servers
               if sock == check
                 break if handle_check
               else
-                pool.wait_until_not_full
+                # uncommenting this may cause 'long tail' response times when all
+                # workers are busy
```

Comment:
I think you may mean to say "when all worker threads are busy".

Reply:
Sorry.  I'm surprised I wrote that, as often the threads in the ThreadPool are called workers, which drives me nuts.

Maybe just "when all threads are busy"?

Reply:
I don't think we should leave in commented out code with a description for what it would do if it's uncommented.

I think the wording "when all threads are busy" is valid, but my preference would be either:

- Delete the code alltogether (older versions are still there in git)
- If you feel strongly it should stay in, then make it user configurable. For example I suggested `enable_keepalives :legacy` as a possible codepath.

I'm fine removing it for now and only adding a config path if there are problems and we need to buy time for a longer term investigation and fix.